### DISCUSSION TITLE
What's the all possible wavefront size for AMD GPUs

### DISCUSSION THREAD

**Discussion by matrixji** [ROCm/ROCm#2318] [Q&A] [answered]
Previously, AIK, Nvidia used 32 warp sizes, and AMD used 64 wavefront sizes.
And I recently noticed that my RX6600 run rocminfo returns:
```
  Wavefront Size:          32(0x20)
  Max Waves Per CU:        32(0x20)
```

So does it mean AMD GPU may have 2 possible wavefronts (32 and 64)? Do we have a spec list to check all possible wavefront size for All AMD GPUs?

**Comment by saadrahim** [comment, marked as answer]
Yes, AMD GPUs have both 64 and 32 wavefront sizes. RDNA GPUs have 32 warp sizes. GPUs may be configurable for 64 warp size but I have to double check on that. @MathiasMagnus and @Naraenda we should add documentation for this.

**Reply by matrixji** [reply]
Thanks for the quick answer, interested in `GPUs may be configurable for 64 warp size`, does it means we could configure wavefront use 64 for all AMD GPUs?

**Reply by matrixji** [reply]
After checking some references, I found the compilation flag `-mwavefrontsize64` is designed for that. :-)
